web/html/src/manager/proxy/container-config.tsx

```diff
@@ -0,0 +1,359 @@
+import * as React from "react";
+
+import { AsyncButton } from "components/buttons";
+import { SubmitButton } from "components/buttons";
+import { Form } from "components/input/Form";
+import { unflattenModel } from "components/input/form-utils";
+import { FormMultiInput } from "components/input/FormMultiInput";
+import { Radio } from "components/input/Radio";
+import { Text } from "components/input/Text";
+import { Panel } from "components/panels/Panel";
+import { TopPanel } from "components/panels/TopPanel";
+import Validation from "components/validation";
+
+import { DEPRECATED_unsafeEquals } from "utils/legacy";
+import Network from "utils/network";
+
+import { ContainerConfigMessages } from "./container-config-messages";
+
+enum SSLMode {
+  UseSSL = "use-ssl",
+  CreateSSL = "create-ssl",
+}
+
+const initialModel = {
+  caCertificate: "",
+  caKey: "",
+  caPassword: "",
+  country: "",
+  state: "",
+  city: "",
+  org: "",
+  orgUnit: "",
+  sslEmail: "",
+  rootCA: "",
+  proxyCertificate: "",
+  proxyKey: "",
+  proxyAdminEmail: "",
+  sslMode: SSLMode.CreateSSL,
+  maxSquidCacheSize: "",
+  proxyFQDN: "",
+  serverFQDN: "",
+  proxyPort: "8022",
+};
+
+function replaceCRLF(content: string | undefined | null) {
+  return content != null ? content.replace(/\r\n/g, "\n") : content;
+}
+
+export function ProxyConfig() {
+  const [messages, setMessages] = React.useState<string[]>([]);
```

Comment:
I think in most views we import `useState` separately and use it directly a-la `useState<string[]>([])`. Would be nice for consistency, but this is the world's smallest nitpick and not really important.